iTechArtLab/Controllers/OrdersController.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using DataAccessLayer.Data;
+using DataAccessLayer.Models;
+using DataAccessLayer.Entities;
+using BuisnessLayer;
+using BuisnessLayer.JWToken;
+using Microsoft.Extensions.Options;
+using Microsoft.AspNetCore.Http;
+using System.Text.Json;
+using Serilog;
+
+namespace iTechArtLab.Controllers
+{
+    [Route("/api/orders")]
+    public class OrdersController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly OrdersManager _ordersManager;
+        private readonly ModelValidator _modelValidator;
+        private readonly JWTokenConfig _jWTokenConfig;
+        private readonly AccessControlManager _accessControlManager;
+        private readonly SessionManager _sessionManager;
+        private readonly JWTokenValidator _jWTokenValidator;
+
+        public OrdersController(ApplicationDbContext context, SessionManager sessionManager, OrdersManager ordersManager,
+            JWTokenValidator jWTokenValidator, AccessControlManager accessControlManager, ModelValidator modelValidator,
+            IOptions<JWTokenConfig> jWTokenOptions)
+        {
+            _context = context;
+            _jWTokenConfig = jWTokenOptions.Value;
+
+            _ordersManager = ordersManager;
+            _modelValidator = modelValidator;
+
+            _sessionManager = sessionManager;
+            _jWTokenValidator = jWTokenValidator;
+            _accessControlManager = accessControlManager;
+        }
+
+        [HttpGet]
+        public async Task<object> GetOrders()
```

Comment:
Change <object> type to IActionResult

Reply:
in case of unathorized user this method returns string with error message and status code 401 that are not IActionResult